src/dbnode/storage/index/results_new_map.go

```diff
@@ -32,17 +34,15 @@ const (
 
 func newResultsMap(idPool ident.Pool) *ResultsMap {
 	return _ResultsMapAlloc(_ResultsMapOptions{
-		hash: func(k ident.ID) ResultsMapHash {
-			return ResultsMapHash(xxhash.Sum64(k.Bytes()))
-		},
-		equals: func(x, y ident.ID) bool {
-			return x.Equal(y)
+		hash: func(k []byte) ResultsMapHash {
+			return ResultsMapHash(xxhash.Sum64(k))
 		},
-		copy: func(k ident.ID) ident.ID {
-			return idPool.Clone(k)
+		equals: bytes.Equal,
+		copy: func(k []byte) []byte {
+			return idPool.Clone(ident.BytesID(k)).Bytes()
```

Comment:
Hm I'd just use a straight up copy here, e.g. `append(make([]byte, 0, len(k)), k...)`

Reply:
Let's just also remove the `idPool` being specified here.

Reply:
(i.e. remove from constructor)